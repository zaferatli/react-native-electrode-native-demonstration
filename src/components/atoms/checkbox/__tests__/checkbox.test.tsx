import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import Checkbox, {CheckboxPropsType} from '../index';
import {Linking} from 'react-native';

const getComponent = (props: CheckboxPropsType) => {
  return <Checkbox {...props} />;
};

it('initial render button', () => {
  renderer.create(getComponent({}));
});

it('Primary Checkbox', () => {
  const props: CheckboxPropsType = {
    icon: 'CheckIcon',
    checked: false,
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Checkbox True', () => {
  const props: CheckboxPropsType = {
    icon: 'CheckIcon',
    checked: true,
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Checkbox With Style', () => {
  const props: CheckboxPropsType = {
    icon: 'CheckIcon',
    checked: false,
    style: {backgroundColor: 'white'},
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});
