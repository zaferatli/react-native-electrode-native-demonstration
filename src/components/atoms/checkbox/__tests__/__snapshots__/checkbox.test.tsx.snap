// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox True 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF0000",
      "borderColor": "#FF0000",
      "borderRadius": 24,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": undefined,
      "height": 24,
      "justifyContent": "center",
      "opacity": 1,
      "width": 24,
    }
  }
>
  <check.svg
    fill="#FFFFFF"
    height={9}
    iconType="CheckIcon"
    width={12}
  />
</View>
`;

exports[`Checkbox With Style 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "#E7E8E9FF",
      "borderRadius": 24,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": undefined,
      "height": 24,
      "justifyContent": "center",
      "opacity": 1,
      "width": 24,
    }
  }
/>
`;

exports[`Primary Checkbox 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#E7E8E9FF",
      "borderRadius": 24,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": undefined,
      "height": 24,
      "justifyContent": "center",
      "opacity": 1,
      "width": 24,
    }
  }
/>
`;
